litho-processor/src/main/java/com/facebook/litho/specmodels/model/EventValidation.java

```diff
@@ -60,14 +64,47 @@
                 "Event declarations must be annotated with @Event."));
       }
 
+      final TypeElement eventRepresentedObject = (TypeElement) eventDeclaration.representedObject;
+      final boolean isKotlinClass = SpecElementTypeDeterminator.determine(eventRepresentedObject)
+          == SpecElementType.KOTLIN_CLASS;
+      final boolean hasOnlyOneNonEmptyCtor = TypeElementUtils
+          .hasOnlyOneNonEmptyConstructor(eventRepresentedObject);
+
+      if (isKotlinClass && !hasOnlyOneNonEmptyCtor) {
+        validationErrors.add(
+            new SpecModelValidationError(
+                eventRepresentedObject,
+                "Event has more than one non-empty constructors. "
+                    + "Prefer to use a `data class` without default values in Kotlin."));
+
+        return validationErrors;
+      }
+
+      final List<String> eventDeclarationGetters =
+          TypeElementUtils.extractGetterMethods(eventRepresentedObject);
+
       for (FieldModel fieldModel : eventDeclaration.fields) {
-        if (!fieldModel.field.modifiers.contains(Modifier.PUBLIC)
-            || (fieldModel.field.modifiers.contains(Modifier.FINAL)
-                && !fieldModel.field.modifiers.contains(Modifier.STATIC))) {
+        final FieldSpec fieldSpec = fieldModel.field;
+        final boolean isPublicField = fieldSpec.modifiers.contains(Modifier.PUBLIC);
+        final boolean isStaticField = fieldSpec.modifiers.contains(Modifier.STATIC);
+
+        final String fieldName = fieldSpec.name;
+        final String possibleFieldGetter = "get" +
+            fieldName.substring(0, 1).toUpperCase() +
+            fieldName.substring(1);
+
+        final boolean fieldHasAvailableGetter =
+            eventDeclarationGetters.contains(possibleFieldGetter);
+
+        final boolean isValidField = (isPublicField || fieldHasAvailableGetter) && !isStaticField;
+
+        if (!isValidField) {
           validationErrors.add(
               new SpecModelValidationError(
                   fieldModel.representedObject,
-                  "Event fields must be declared as public non-final."));
+                  "Event fields must be declared as public non-final, "
```

Comment:
Should we allow `public` fields only for empty constructors? Let's keep this only for old-style's support

Reply:
Sure will update it!